lib/beaker/puppet_install_helper.rb

```diff
@@ -83,9 +83,12 @@ def run_puppet_install_helper_on(hosts, type_arg = find_install_type, version =
         end
       end
     when 'agent'
-      if find_agent_sha.nil?
+      if ENV['BEAKER_PUPPET_COLLECTION'] =~ /-nightly$/
         # Workaround for RE-10734
-        options[:release_apt_repo_url] = "http://apt.puppetlabs.com/#{ENV['BEAKER_PUPPET_COLLECTION']}" if ENV['BEAKER_PUPPET_COLLECTION'] =~ /-nightly$/
+        options[:release_apt_repo_url] = "http://apt.puppetlabs.com/#{ENV['BEAKER_PUPPET_COLLECTION']}"
+        options[:win_download_url] = 'http://nightlies.puppet.com/downloads/windows'
```

Comment:
where is `win_download_url` used?

Reply:
what about ios? apple packages also don't have a release package you can install, similar to windows. Do you test against mac platforms?

Reply:
The win_download_url and mac_download_url defaults are set at https://github.com/puppetlabs/beaker-puppet/blob/63ea32a0d7caa8f261c533b020625de19569f971/lib/beaker-puppet/install_utils/foss_defaults.rb#L11-L12 but I need to override those when calling install_puppet_agent_on() below.

Reply:
Added the mac_download_url also, though I don't think we test on macos anywhere.